single.php

```diff
@@ -63,26 +63,7 @@
 
 // Build the shortcode for articles block.
 if ('yes' === $post->include_articles) {
-    $tag_id_array = [];
-    foreach ($post->tags() as $post_tag) {
-        $tag_id_array[] = $post_tag->id;
-    }
-    $category_id_array = [];
-    foreach ($post->terms('category') as $category) {
-        $category_id_array[] = $category->id;
-    }
-
-    $block_attributes = [
-        'exclude_post_id' => $post->ID,
-        'tags' => $tag_id_array,
-        'categories' => $category_id_array,
-        'article_heading' => __('Related Articles', 'planet4-blocks'),
-        'read_more_text' => __('Load more', 'planet4-blocks'),
-    ];
-
-    $post->articles = '<!-- wp:planet4-blocks/articles '
-        . wp_json_encode($block_attributes, JSON_UNESCAPED_SLASHES)
-        . ' /-->';
+    $post->articles = '<!-- wp:planet4-blocks/posts-list-test /-->';
```

Comment:
Can't we just pass here a query block code with the "post-list" variation?
Something like:

<details>
<summary>Click to expand!</summary>

``` 
<!-- wp:query {"queryId":4,"query":{"perPage":3,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[1113],"sticky":"","inherit":false,"postIn":[],"hasPassword":false},"namespace":"planet4-blocks/posts-list","className":"posts-list p4-query-loop is-custom-layout-list","layout":{"type":"default","columnCount":3}} -->
<div class="wp-block-query posts-list p4-query-loop is-custom-layout-list"><!-- wp:group {"layout":{"type":"flex","justifyContent":"space-between"}} -->
<div class="wp-block-group"><!-- wp:heading {"lock":{"move":true}} -->
<h2 class="wp-block-heading">Related Posts</h2>
<!-- /wp:heading -->

<!-- wp:navigation-link {"label":"See all stories","url":"http://www.planet4.test/news-stories/","className":"see-all-link"} /--></div>
<!-- /wp:group -->

<!-- wp:paragraph {"placeholder":"Enter description","lock":{"move":true},"style":{"spacing":{"margin":{"top":"24px","bottom":"36px"}}}} -->
<p style="margin-top:24px;margin-bottom:36px"></p>
<!-- /wp:paragraph -->

<!-- wp:query-no-results -->
<!-- wp:paragraph -->
<p>No posts found. (This default text can be edited)</p>
<!-- /wp:paragraph -->
<!-- /wp:query-no-results -->

<!-- wp:post-template {"lock":{"move":true,"remove":true}} -->
<!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:post-featured-image {"isLink":true} /-->

<!-- wp:group -->
<div class="wp-block-group"><!-- wp:group {"layout":{"type":"flex"}} -->
<div class="wp-block-group"><!-- wp:post-terms {"term":"category","separator":" | "} /-->

<!-- wp:post-terms {"term":"post_tag","separator":" "} /--></div>
<!-- /wp:group -->

<!-- wp:post-title {"isLink":true} /-->

<!-- wp:post-excerpt /-->

<!-- wp:group {"className":"posts-list-meta"} -->
<div class="wp-block-group posts-list-meta"><!-- wp:post-author-name {"isLink":true} /-->

<!-- wp:post-date /--></div>
<!-- /wp:group --></div>
<!-- /wp:group --></div>
<!-- /wp:columns -->
<!-- /wp:post-template -->

<!-- wp:buttons {"lock":{"move":true},"className":"carousel-controls","layout":{"type":"flex","justifyContent":"space-between","orientation":"horizontal","flexWrap":"nowrap"}} -->
<div class="wp-block-buttons carousel-controls"><!-- wp:button {"className":"carousel-control-prev"} -->
<div class="wp-block-button carousel-control-prev"><a class="wp-block-button__link wp-element-button">Prev</a></div>
<!-- /wp:button -->

<!-- wp:button {"className":"carousel-control-next"} -->
<div class="wp-block-button carousel-control-next"><a class="wp-block-button__link wp-element-button">Next</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons -->

<!-- wp:navigation-link {"label":"See all stories","url":"http://www.planet4.test/news-stories/","className":"see-all-link"} /--></div>
<!-- /wp:query -->
```

Reply:
Right, it's a lot of code. I'll keep digging...

Reply:
Any way to import a JSON file into both the `single.php` and the assets/src/blocks/PostsListTest/template.js` to avoid repeating the block attributes and inner blocks?

Reply:
@mardelnet, @mleray @sagarsdeshmukh, is it possible to have a partial that takes arguments/params that way we could modify this [listing page](https://github.com/greenpeace/planet4-master-theme/blob/main/parts/query-listing-page.html) or create one just for the Related Articles.

Reply:
Hm 🤔 This would still be duplicated code indeed, but it would make it more readable so I would say it's better!

Reply:
Sorry just looking into this now @mleray and @mardelnet. Question, does the new template part also accept params for how to display the post lists block, because the data needed to display each block is different depending on the attributes.

Reply:
@Osong-Michael 
I think we cannot pass parameters to a template part. Our options are:
- Turn the template into a custom block that accepts attributes.
- Create a filter to modify the content of the template part (for example, [this one](https://developer.wordpress.org/reference/hooks/render_block_this-name/))